packages/terra-dropdown-button/src/Button.jsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import styles from './Button.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The text that is displayed on the option.
+   */
+  label: PropTypes.string.isRequired,
+  /**
+   * The function that is called when the option is clicked or enter/space is pressed with it focused.
+   */
+  onClick: PropTypes.func.isRequired,
+  /**
+   * @private
+   * Whether or not the option should show active styles
+   */
+  isActive: PropTypes.bool,
+  /**
+   * @private
+   * Whether or not the option should show keyboard focus styles
+   */
+  isFocused: PropTypes.bool,
+  /**
+   * @private
+   * Callback to tell the parent it should close the dropdown
+   */
+  requestClose: PropTypes.func,
+};
+
+const Option = ({
+  label, onClick, isFocused, isActive, requestClose, ...customProps
+}) => (
+  <li>
+    <div
+      {...customProps}
+      onMouseUp={() => { onClick(); requestClose(); }}
```

Comment:
This is a little tricky, using the default example as a reference. 

If you open the dropdown then click and hold option 1, drag the mouse down to option 4 ( or any other option ) and then release the mouse, that option will fire a selection. This might lead to some unexpected user actions, it's not the option that you clicked that fires, but whichever one you happen to mouse up on. 

If you were trying to mimic some of the functionality of the Select, the Select records which button was pressed on mouse down and then compares that button against the button that receives the mouse up event. If they don't match it doesn't make a selection.

https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/_Menu.jsx#L446

Reply:
Not sure why I was using `onMouseUp`, addressed in f3b73577383099a9bddb5383f6a820868611af97